cmd/csi-proxy-api-gen/generators/client_generated_gen.go

```diff
@@ -56,7 +58,19 @@ type Client struct {
 // NewClient returns a client to make calls to the $.groupName$ API group version $.version$.
 // It's the caller's responsibility to Close the client when done.
 func NewClient() (*Client, error) {
-	pipePath := client.PipePath(groupName, version)
+	pipePath := client.PipePath(GroupName, Version)
+	return NewClientWithPipePath(pipePath)
+}
+
+// NewClientWithPipePath returns a client to make calls to the named pipe located at "pipePath".
+// It's the caller's responsibility to Close the client when done.
+func NewClientWithPipePath(pipePath string) (*Client, error) {
+
+	// verify that the pipe exists
+	_, err := winio.DialPipe(pipePath, nil)
```

Comment:
wondering what is the difference between winio.DialPipe and the following grpc.Dial?

Reply:
I read the first test in https://github.com/microsoft/go-winio/blob/master/pipe_test.go, `grpc.Dial` creates connections in the background unless `WithBlock()` is also sent as an option (which I also tried but got stuck in the v1 request)